Userland/useradd.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019-2020, Jesse Buhagiar <jooster669@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <AK/String.h>
+#include <LibCore/CArgsParser.h>
+#include <ctype.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+constexpr uid_t BASE_UID = 1000;
+constexpr gid_t USERS_GID = 100;
+constexpr const char* DEFAULT_SHELL = "/bin/Shell";
+
+int main(int argc, char** argv)
+{
+    const char* home_path = nullptr;
+    int uid = 0;
+    int gid = USERS_GID;
+    bool create_home_dir = false;
+    const char* shell = DEFAULT_SHELL;
+    const char* gecos = "";
+    const char* username = nullptr;
+
+    CArgsParser args_parser;
+    args_parser.add_option(home_path, "Home directory for the new user", "home-dir", 'd', "path");
+    args_parser.add_option(uid, "User ID (uid) for the new user", "uid", 'u', "uid");
+    args_parser.add_option(gid, "Group ID (gid) for the new user", "gid", 'g', "gid");
+    args_parser.add_option(create_home_dir, "Create home directory if it does not exist", "create-home", 'm');
+    args_parser.add_option(shell, "Name of default shell binary for the new user", "shell", 's', "path-to-shell");
```

Comment:
And the help string here should say "path", not "name", too.

E.g. it's not "bash", it's "`/usr/local/bin/bash`"